### PR TITLE
IGNITE-16227 .NET: Thin 3.0: Add RecordView with POCO mapping

### PR DESCRIPTION
**Pull request by ptupitsyn**
* Add `ITable.GetRecordView<T>` to map data to arbitrary user objects
* Rename `ITable.RecordView` to `ITable.RecordBinaryView` to avoid clashing with `GetRecordView` (see [CA1721: Property names should not match get methods](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1721))
* Extract generic `RecordView` implementation that works with tuples and user objects.
* Implement basic reflection-based property mapping. Proper high-perf mapping with field support to be done separately: [IGNITE-16341](https://issues.apache.org/jira/browse/IGNITE-16341).

### CODE REVIEW

File: modules/platforms/dotnet/Apache.Ignite/Internal/Table/Serialization/ObjectSerializerHandler.cs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+            {
+                var col = columns[i];
+                var prop = GetPropertyIgnoreCase(type, col.Name);
+
+                if (i < schema.KeyColumnCount)
+                {
+                    if (prop != null)
+                    {
+                        prop.SetValue(res, prop.GetValue(key));
+                    }
```
Comment (isapego) [inline modules/platforms/dotnet/Apache.Ignite/Internal/Table/Serialization/ObjectSerializerHandler.cs:88]: Here we just silently ignore a case when user class does not have a column that present in schema. Is this behavior discussed somewhere?

**Review by isapego — APPROVED** [review]
Overall looks good. See my few comments.

File: modules/platforms/dotnet/Apache.Ignite/Internal/Table/Serialization/RecordSerializer.cs [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        /// <param name="buf">Buffer.</param>
+        /// <param name="schema">Schema.</param>
+        /// <param name="key">Key part.</param>
+        /// <returns>Resulting record with key and value parts.</returns>
+        public T? ReadValue(PooledBuffer buf, Schema? schema, T key)
+        {
+            if (schema == null)
+            {
+                return null;
+            }
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/platforms/dotnet/Apache.Ignite/Internal/Table/Serialization/RecordSerializer.cs:62]: Null schema means null record in our protocol (e.g. key does not exist in `table.get`). I've added comments to clarify this.
There is no case of "missing schema".

File: modules/platforms/dotnet/Apache.Ignite/Internal/Table/Serialization/ObjectSerializerHandler.cs (hunk 1 above)
Comment (ptupitsyn) [reply modules/platforms/dotnet/Apache.Ignite/Internal/Table/Serialization/ObjectSerializerHandler.cs:88]: Same behavior as in Java. I could not find any discussions on this topic, but it seems reasonable to me.

File: modules/platforms/dotnet/Apache.Ignite/Internal/Table/Serialization/RecordSerializer.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (isapego) [reply modules/platforms/dotnet/Apache.Ignite/Internal/Table/Serialization/RecordSerializer.cs]: Understood

File: modules/platforms/dotnet/Apache.Ignite/Internal/Table/Serialization/ObjectSerializerHandler.cs (hunk 1 above)
Comment (isapego) [reply modules/platforms/dotnet/Apache.Ignite/Internal/Table/Serialization/ObjectSerializerHandler.cs:88]: Ok, but maybe we still need to discuss that.

**Pull request merged**